### PR TITLE
Add test to ensure we don't crash upon upstream error

### PR DESCRIPTION
**Pull request by henridf**
zq previously crashed when run with in a shell pipeline with an error. The issue was incidentally fixed in eeb41b0c25cacb4e2b53e69427c69a5917b9eea5, and this commit adds a test to ensure we don't regress.

### CODE REVIEW

File: tests/suite/errors/zat.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"github.com/mccanne/zq/pkg/test"
+)
+
+var Exec = test.Exec{
+	Name:     "error upstream",
+	Command:  `zat simple.zng | zq -`,
+	Input:    test.Trim(input),
+	Expected: "",
+}
```
Comment (nwt) [inline tests/suite/errors/zat.go]: This is really a test of how zq handles empty input, and I think you should dress it as such.

It's also a test we should do for every "zq -f" format, but I don't think that needs to hold up this PR.

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,7 @@ fmt:
 test-unit:
 	@go test -short ./...
 
-test-system: build
+test-system: install
```
Comment (nwt) [inline Makefile]: If you're going to do this, you should get rid of everything related to the dist directory, including all the PATH stuff in pkg/test. And I think that deserves a separate PR.

File: tests/suite/errors/zat.go (hunk 1 above)
Comment (henridf) [reply tests/suite/errors/zat.go]: done

File: Makefile (hunk 2 above)
Comment (henridf) [reply Makefile]: this was a misguided attempt to do something else.  removed the commit.

Comment (nwt) [reply Makefile]: 👍

File: tests/suite/errors/emptyinput.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package errors
+
+import (
+	"github.com/mccanne/zq/pkg/test"
+)
+
+var Exec = test.Exec{
+	Name:     "empty input",
+	Command:  `echo "" | zq -`,
+	Input:    test.Trim(input),
```
Comment (nwt) [inline tests/suite/errors/emptyinput.go]: Nit: Maybe simplify to this?
```go
Command: "zq -",
Input:   "",
```

Comment (henridf) [reply tests/suite/errors/emptyinput.go]: better!
